File: client-sdk/csharp/sdk/src/VertexCache/Sdk/Protocol/Parsers/NilResponseParser.cs
using VertexCache.Sdk.Results;

namespace VertexCache.Sdk.Protocol.Parsers
{
    public class NilResponseParser : IResponseParser
    {
        public bool CanParse(string response) => response == "+(nil)";

        public VCacheResult Parse(string response) => VCacheResult.SuccessWithRaw(response);
    }
}

File: client-sdk/go/sdk/protocol/parsers/nil_response_parser.go
package parsers

import "vertexcache/sdk/results"

type NilResponseParser struct{}

func (p *NilResponseParser) CanParse(response string) bool {
	return response == "+(nil)"
}

func (p *NilResponseParser) Parse(response string) *results.VCacheResult {
	return results.NewNil()
}
